app/lib/actions/katello/repository/create.rb

```diff
@@ -12,38 +12,42 @@ def plan(repository, clone = false, plan_create = false)
-                                        name: repository.name,
-                                        docker_upstream_name: repository.docker_upstream_name,
-                                        feed: repository.url,
-                                        ssl_ca_cert: certs[:ssl_ca_cert],
-                                        ssl_client_cert: certs[:ssl_client_cert],
-                                        ssl_client_key: certs[:ssl_client_key],
-                                        unprotected: repository.unprotected,
-                                        checksum_type: repository.checksum_type,
-                                        path: path,
-                                        download_policy: repository.download_policy,
-                                        with_importer: true,
-                                        mirror_on_sync: repository.mirror_on_sync?,
-                                        ssl_validation: certs[:ssl_validation],
-                                        upstream_username: repository.upstream_username,
-                                        upstream_password: repository.upstream_password)
-
-            return if create_action.error
+            unless repository.pulp_repo_facts
+              certs = repository.importer_ssl_options
+              create_action = plan_action(create_action,
+                                          content_type: repository.content_type,
+                                          pulp_id: repository.pulp_id,
+                                          name: repository.name,
+                                          docker_upstream_name: repository.docker_upstream_name,
+                                          feed: repository.url,
+                                          ssl_ca_cert: certs[:ssl_ca_cert],
+                                          ssl_client_cert: certs[:ssl_client_cert],
+                                          ssl_client_key: certs[:ssl_client_key],
+                                          unprotected: repository.unprotected,
+                                          checksum_type: repository.checksum_type,
+                                          path: path,
+                                          download_policy: repository.download_policy,
+                                          with_importer: true,
+                                          mirror_on_sync: repository.mirror_on_sync?,
+                                          ssl_validation: certs[:ssl_validation],
+                                          upstream_username: repository.upstream_username,
+                                          upstream_password: repository.upstream_password)
 
+              return if create_action.error
+            end
             # when creating a clone, the following actions are handled by the
             # publish/promote process
+            has_content = repository.content.present? rescue false
```

Comment:
What is the rescue here for? Is it because `repository.content` could be nil? In that case, it would be better to use `try(:content)` I think (rather than catching all possible exceptions).